cluster/connection_strategy_config_test.go

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cluster_test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/hazelcast/hazelcast-go-client/cluster"
+)
+
+func TestReconnectMode_MarshalText(t *testing.T) {
+	t.Logf("reconnection-mode on")
+	rmOn := cluster.ReconnectModeOn
+	text, err := rmOn.MarshalText()
+	if err != nil {
+		t.Error(err)
+	}
+	require.Equal(t, []byte(`on`), text)
+	t.Logf("reconnection-mode off")
+	rmOff := cluster.ReconnectModeOff
+	text, err = rmOff.MarshalText()
+	if err != nil {
+		t.Error(err)
+	}
+	require.Equal(t, []byte(`off`), text)
+	t.Logf("reconnection-mode invalid")
+	rmInvalid := cluster.ReconnectModeOff + 1
+	_, err = rmInvalid.MarshalText()
+	require.Error(t, err, "should be an error")
+}
```

Comment:
Fixed: [88b6f60](https://github.com/hazelcast/hazelcast-go-client/pull/809/commits/88b6f60a915c3d0237ae5f1fb97bdbda52cedfbd)

Reply:
The new code has this:
```
got, err := tc.rm.MarshalText()
if tc.hasErr {
    assert.Error(t, err)
} else {
    assert.Equal(t, tc.expected, got)
}
```
What if `err != nil` ? Although the test will fail, we won't know it failed because there was an error. How about this:
```
got, err := tc.rm.MarshalText()
if err != nil {
    if tc.hasErr {
	return
    }
    t.Fatal(err)
}
assert.Equal(t, tc.expected, got)

```

Reply:
Thanks, fixed [74c3f9d](https://github.com/hazelcast/hazelcast-go-client/pull/809/commits/74c3f9dd5fc4c20247ec31de821b2bc005d03252).